lib/endpoints/class-wp-json-posts-controller.php

```diff
@@ -639,6 +639,20 @@ protected function prepare_item_for_database( $request ) {
 			if ( ! current_user_can( $post_type->cap->publish_posts ) ) {
 				return new WP_Error( 'json_forbidden', __( 'Sorry, you are not allowed to create password protected posts in this post type' ), array( 'status' => 403 ) );
 			}
+
+			if ( ! empty( $schema['properties']['sticky'] ) && ! empty( $request['sticky'] ) ) {
```

Comment:
Where is the password conditional in this statement?

Also should check that an already sticky post can't be assigned a password.